src/Workspaces/Core/Portable/Utilities/SerializableBytes.cs

```diff
@@ -244,6 +245,12 @@ public byte[] ToArray()
                 return array;
             }
 
+            public ImmutableArray<byte> ToImmutableArray()
+            {
+                var array = ToArray();
+                return ImmutableArrayExtensions.DangerousCreateFromUnderlyingArray(ref array);
```

Comment:
>return ImmutableArrayExtensions.DangerousCreateFromUnderlyingArray(ref array) [](start = 16, length = 77)

Why not use ```ImmutableArray<T>.Builder```. Create a builder of the right size ```ImmutableArray.CreateBuilder<T>(size)```, fill it with data and call ```MoveToImmutable```.

Reply:
Because I'd need to rewrite all the code that works with the mutable arrays.